pkg/promtail/client/client.go

```diff
@@ -349,30 +340,17 @@ func (c *client) getTenantID(labels model.LabelSet) string {
 
 // Stop the client.
 func (c *client) Stop() {
-	c.once.Do(func() { close(c.quit) })
+	close(c.entries)
```

Comment:
Is there a reason to drop `once`? Or are we considering it a bug if `Stop` is called multiple times?

Reply:
I'd like to leave the once call in unless there's a distinct reason not to. I'd rather propagate an error than panic, although this may already be guaranteed due to how this is used, I'm not sure.

Reply:
yes let's keep the once.